src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/ProjectEngineFactory_3_0.cs

```diff
@@ -1,9 +1,17 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Razor.Language;
+using System;
+
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Common;
 
 internal class ProjectEngineFactory_3_0 : ProjectEngineFactory
 {
     protected override string AssemblyName { get; } = "Microsoft.AspNetCore.Mvc.Razor.Extensions";
+
+    public override RazorProjectEngine Create(
+        RazorConfiguration configuration,
+        RazorProjectFileSystem fileSystem,
+        Action<RazorProjectEngineBuilder> configure) => Create(configuration, fileSystem, configure, registerCompilerFeatures: true);
```

Comment:
This is the main part of the change (`registerCompilerFeatures: true`).

Reply:
Good catch, I'll be this was really annoying to track down.